tools/build/wheel_linux_before_all.sh

```diff
@@ -1,25 +1,27 @@
-#!/bin/bash
+#!/bin/sh
 
 # Configure the libraries needed to build wheel packages on linux.
 # This script is designed to be used by cibuildwheel as CIBW_BEFORE_ALL_LINUX
 
-set -euo pipefail
-set -x
+set -xe
 
-dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+. /etc/os-release
 
-# Install libpq
-# Note that the pgdg doesn't have an aarch64 repository so wheels are build
-# with the libpq packaged with Debian 9, which is 9.6.
-if [[ ! "$AUDITWHEEL_ARCH" = "aarch64" ]]; then
-    source /etc/os-release
-    echo "deb http://apt.postgresql.org/pub/repos/apt ${VERSION_CODENAME}-pgdg main" \
-        > /etc/apt/sources.list.d/pgdg.list
-    # TODO: on 2021-09-30 curl fails with
-    # curl: (60) SSL certificate problem: certificate has expired
-    # Test again later if -k can be removed.
-    curl -sk https://www.postgresql.org/media/keys/ACCC4CF8.asc \
-        | apt-key add -
-fi
-apt-get update
-apt-get -y install libpq-dev
+# Install PostgreSQL development files.
+case "$ID" in
+    alpine)
+        apk add --no-cache postgresql-dev
+        ;;
+    debian)
+        # Note that the pgdg doesn't have an aarch64 repository so wheels are
+        # build with the libpq packaged with Debian 9, which is 9.6.
+        if [ "$AUDITWHEEL_ARCH" != 'aarch64' ]; then
+            echo "deb http://apt.postgresql.org/pub/repos/apt $VERSION_CODENAME-pgdg main" \
+                > /etc/apt/sources.list.d/pgdg.list
+            curl -sf https://www.postgresql.org/media/keys/ACCC4CF8.asc \
```

Comment:
What is `-f` for? It seems that it makes curl fail silently. Don't we want to fail noisely?

Reply:
The documentation is actually poorly phrased here in my opinion. Normally, curl only exists with non-zero if there's a network related error. But if the server returns a 4xx/5xx status code, that's considered "ok".

What the `-f` flag does is change this behaviour, so a 4xx/5xx (maybe others as well?) results in the exit code `22`.

```console
$ curl -i --http1.1 https://httpbin.org/status/500
HTTP/1.1 500 INTERNAL SERVER ERROR
Date: Mon, 08 Nov 2021 16:33:28 GMT
Content-Type: text/html; charset=utf-8
Content-Length: 0
Connection: keep-alive
Server: gunicorn/19.9.0
Access-Control-Allow-Origin: *
Access-Control-Allow-Credentials: true

$ echo $?
0
```
```console
$ curl -if --http1.1 https://httpbin.org/status/500
HTTP/1.1 500 INTERNAL SERVER ERROR
Date: Mon, 08 Nov 2021 16:33:37 GMT
Content-Type: text/html; charset=utf-8
Content-Length: 0
Connection: keep-alive
Server: gunicorn/19.9.0
Access-Control-Allow-Origin: *
Access-Control-Allow-Credentials: true

curl: (22) The requested URL returned error: 500
$ echo $?
22
```

Reply:
Ok, that seems more sensible indeed.

Reply:
I am not sure this is working as expected: https://github.com/psycopg/psycopg/runs/4145304671?check_suite_focus=true#step:6:216